cli/determined_cli/agent.py

```diff
@@ -22,19 +21,25 @@ def list_agents(args: argparse.Namespace) -> None:
     r = api.get(args.master, "agents")
 
     agents = r.json()
-    headers = ["Agent Name", "Registered Time", "Slots", "Containers", "Label"]
-    values = [
-        [
-            local_id(agent_id),
-            render.format_time(agent["registered_time"]),
-            len(agent["slots"]),
-            agent["num_containers"],
-            agent["label"],
-        ]
+    agents = [
+        {
+            "id": local_id(agent_id),
+            "registered_time": agent["registered_time"],
+            "num_slots": len(agent["slots"]),
+            "num_containers": agent["num_containers"],
+            "label": agent["label"],
+        }
         for agent_id, agent in sorted(agents.items())
     ]
 
-    print(tabulate.tabulate(values, headers, tablefmt="presto"), flush=False)
+    if args.json:
+        print(json.dumps(agents, indent=4))
+        return
+
+    headers = ["Agent Name", "Registered Time", "Slots", "Containers", "Label"]
+    values = [a.values() for a in agents]
```

Comment:
blocking: This is relying on the dict order being preserved, so it's incorrect on Python 3.5, which we still claim to support. (And the ordering guarantee is technically an implementation detail on 3.6, but I'm alright with not worrying about non-CPython implementations.)

(Not that I _want_ us to keep supporting 3.5, but as long as we do...)

Reply:
Very good catch!!!!

This code occur to be as I was implementing it, but I forgot that we can't rely on dict ordering. Regardless, dict ordering is a pretty subtle property to be depending on (and I hate that feature of Python anyway) -- so I shouldn't have done so in any case.

Fixed by switching to `OrderedDict`.